File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/LoggedModelCollection.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;



/**
 * Result of listing logged-models.
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=LoggedModelCollection.Builder.class)

public final class LoggedModelCollection  {
    @Deprecated
    @java.beans.ConstructorProperties({"models", "nextPageToken"})
    public LoggedModelCollection(java.util.List<LoggedModel> models, String nextPageToken) {
        super();
        this.models = models;
        this.nextPageToken = nextPageToken;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * LoggedModels that match the search criteria
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("models")
private java.util.List<LoggedModel> models;

        /**
         * LoggedModels that match the search criteria
         * @param models the value to set
         * @return this builder
         **/
        

public Builder models(java.util.List<LoggedModel> models) {
    this.models = models;
    return this;
}
            /**
     * Token that can be used to retrieve the next page of logged-models. An empty token means that no more logged-models are available for retrieval.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("next_page_token")
private String nextPageToken;

        /**
         * Token that can be used to retrieve the next page of logged-models. An empty token means that no more logged-models are available for retrieval.
         * @param nextPageToken the value to set
         * @return this builder
         **/
        

public Builder nextPageToken(String nextPageToken) {
    this.nextPageToken = nextPageToken;
    return this;
}


        public LoggedModelCollection build() {
            LoggedModelCollection model = new LoggedModelCollection(this.models
                , this.nextPageToken);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(LoggedModelCollection model) {
                this.models(model.getModels());
    this.nextPageToken(model.getNextPageToken());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * LoggedModels that match the search criteria
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("models")
    private final java.util.List<LoggedModel> models;

        /**
     * LoggedModels that match the search criteria
     * @return the value
     **/
    
    public java.util.List<LoggedModel> getModels() {
        return models;
    }


        /**
     * Token that can be used to retrieve the next page of logged-models. An empty token means that no more logged-models are available for retrieval.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("next_page_token")
    private final String nextPageToken;

        /**
     * Token that can be used to retrieve the next page of logged-models. An empty token means that no more logged-models are available for retrieval.
     * @return the value
     **/
    
    public String getNextPageToken() {
        return nextPageToken;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("LoggedModelCollection(");
        sb.append("models=").append(String.valueOf(this.models));
        sb.append(", nextPageToken=").append(String.valueOf(this.nextPageToken));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof LoggedModelCollection)) {
            return false;
        }

        LoggedModelCollection other = (LoggedModelCollection) o;
        return java.util.Objects.equals(this.models, other.models) &&
            java.util.Objects.equals(this.nextPageToken, other.nextPageToken);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.models == null ? 43 : this.models.hashCode());
        result = (result * PRIME) + (this.nextPageToken == null ? 43 : this.nextPageToken.hashCode());
        return result;
    }


}

File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/logged_model_collection.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class LoggedModelCollection(object):
    """
    Result of listing logged-models.
    """

    def __init__(self, **kwargs):
        """
        Initializes a new LoggedModelCollection object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param models:
            The value to assign to the models property of this LoggedModelCollection.
        :type models: list[oci.aidataplatform_dp.models.LoggedModel]

        :param next_page_token:
            The value to assign to the next_page_token property of this LoggedModelCollection.
        :type next_page_token: str

        """
        self.swagger_types = {
            'models': 'list[LoggedModel]',
            'next_page_token': 'str'
        }

        self.attribute_map = {
            'models': 'models',
            'next_page_token': 'next_page_token'
        }

        self._models = None
        self._next_page_token = None

    @property
    def models(self):
        """
        Gets the models of this LoggedModelCollection.
        LoggedModels that match the search criteria


        :return: The models of this LoggedModelCollection.
        :rtype: list[oci.aidataplatform_dp.models.LoggedModel]
        """
        return self._models

    @models.setter
    def models(self, models):
        """
        Sets the models of this LoggedModelCollection.
        LoggedModels that match the search criteria


        :param models: The models of this LoggedModelCollection.
        :type: list[oci.aidataplatform_dp.models.LoggedModel]
        """
        self._models = models

    @property
    def next_page_token(self):
        """
        Gets the next_page_token of this LoggedModelCollection.
        Token that can be used to retrieve the next page of logged-models. An empty token means that no more logged-models are available for retrieval.


        :return: The next_page_token of this LoggedModelCollection.
        :rtype: str
        """
        return self._next_page_token

    @next_page_token.setter
    def next_page_token(self, next_page_token):
        """
        Sets the next_page_token of this LoggedModelCollection.
        Token that can be used to retrieve the next page of logged-models. An empty token means that no more logged-models are available for retrieval.


        :param next_page_token: The next_page_token of this LoggedModelCollection.
        :type: str
        """
        self._next_page_token = next_page_token

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other
